### PR TITLE
Introduce support for titles

### PR DESCRIPTION
**Pull request by mixonic**
Fixes much but not all of #94 (some of those screens are still in progress). I've added a checklist there.

This introduces support for titles. Each route can have a `title` property. After transition, the route tree will be walked upward until a title string or function is found. This function is then called.

There is also support for building URLs with tokens. I'm going to look at a refactor using this logic now that the tests are in place.

WIP, please do not merge.

### CODE REVIEW

**Comment by mixonic** [comment]
@bantic for you to review and merge.

File: app/utils/title-route-extensions.js [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+  // Provided by Ember
+  _actions: {
+    collectTitleTokens: function(tokens) {
+      var titleToken = this.titleToken;
+      if (typeof this.titleToken === 'function') {
+        titleToken = this.titleToken(this.currentModel);
+      }
+
+      if (Ember.isArray(titleToken)) {
+        tokens.unshift.apply(this, titleToken);
```
Comment (bantic) [inline app/utils/title-route-extensions.js]: is there a performance benefit to doing it this way instead of `tokens = titleToken.concat(tokens)` ?

Comment (bantic) [inline app/utils/title-route-extensions.js]: and I may be reading this wrong but wouldn't it be `tokens.unshift.apply(tokens, titleToken)`?

**Comment by bantic** [comment]
It's funny how cool watching a page's title change dynamically can be. Neat to see this in action. The use of 'token' throws me off a little — I would think that would be named like 'breadCrumbs' or something. I wonder if this is a more complicated/comprehensive solution than we strictly need, but it is likely be better suited for future use cases than a simpler system.

I had a question about `tokens.unshift.apply` above — after double-checking me on that, please merge at your leisure.

File: app/utils/title-route-extensions.js (hunk 1 above)
Comment (mixonic) [inline app/utils/title-route-extensions.js]: Using `concat` would allocate a second array. I agree this should be `apply(tokens` though, confirming.

**Pull request merged**